Crash report 4.0-rc-3: NPE in LegacyEditorFragment.onActivityResult

```
java.lang.RuntimeException: Failure delivering result ResultInfo{who=android:fragment:0, request=4, result=0, data=null} to activity {org.wordpress.android/org.wordpress.android.ui.posts.EditPostActivity}: java.lang.NullPointerException
       at android.app.ActivityThread.deliverResults(ActivityThread.java:3532)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:3575)
       at android.app.ActivityThread.access$1100(ActivityThread.java:162)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1416)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:194)
       at android.app.ActivityThread.main(ActivityThread.java:5410)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.NullPointerException
       at org.wordpress.android.editor.LegacyEditorFragment.onActivityResult(LegacyEditorFragment.java:325)
       at android.app.Activity.dispatchActivityResult(Activity.java:5315)
       at android.app.ActivityThread.deliverResults(ActivityThread.java:3528)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:3575)
       at android.app.ActivityThread.access$1100(ActivityThread.java:162)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1416)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:194)
       at android.app.ActivityThread.main(ActivityThread.java:5410)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

--- a/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/LegacyEditorFragment.java
+++ b/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/LegacyEditorFragment.java
@@ -321,7 +321,7 @@ public class LegacyEditorFragment extends EditorFragmentAbstract implements Text
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
 
-        if (requestCode == LegacyEditorFragment.ACTIVITY_REQUEST_CODE_CREATE_LINK) {
+        if (requestCode == LegacyEditorFragment.ACTIVITY_REQUEST_CODE_CREATE_LINK && data != null) {
             Bundle extras = data.getExtras();
             if (extras == null) {
                 return;

